src/worker/connectionstate.js

```diff
@@ -16,6 +16,8 @@ class IrcBuffer {
         this.key = '';
         this.joined = false;
         this.topic = '';
+        this.modes = '';
```

Comment:
We should make this a `Object.create(null)` instance so that it can only ever store one instance of a mode at a time and also supports storing mode params